x/btccheckpoint/README.md

```diff
@@ -0,0 +1,222 @@
+# BTCCheckpoint
+
+The BTC Checkpoint module is responsible for receiving and managing the checkpoints of Babylon's state on the Bitcoin blockchain. This includes:
+- handling requests for submitting raw checkpoints
+- processing Bitcoin SPV proofs for submitted checkpoints
+- managing the lifecycle of checkpoints (SEALED, SUBMITTED, CONFIRMED, FINALIZED)
+- handling the verification and finalization of checkpoints
+- distributing rewards for successful checkpoint submissions, and
+- proactively updating the status of checkpoints based on Bitcoin blockchain confirmations
+
+## Table of contents
+
+- [Table of contents](#table-of-contents)
+- [Concepts](#concepts)
+- [States](#states)
+  - [Parameters](#parameters)
+  - [Epoch data](#epoch-data)
+  - [Latest Finalized Epoch](#latest-finalized-epoch)
+  - [Submission data](#submission-data)
+  - [Transient States](#transient-states)
+- [Messages](#messages)
+  - [MsgInsertBTCSpvProof](#msginsertbtcspvproof)
+  - [MsgUpdateParams](#msgupdateparams)
+- [EndBlocker](#endblocker)
+- [Queries](#queries)
+
+## Concepts 
+Babylon's BTC Checkpoint module allows Babylon chain to periodically checkpoint its state onto the onto the Bitcoin blockchain whilst simultaneously verifying the checkpoints. The process involves two main components:
+
+Checkpoint Submission:
+1. When a new checkpoint for a specific epoch is available, the [vigilante](https://github.com/babylonlabs-io/vigilante) collects the necessary proof from the Bitcoin blockchain. 
+2. This proof is called an `SPVProof` (Simplified Payment Verification Proof), consisting of the Bitcoin transaction, Index of the transaction, the Merkle path and the Bitcoin header that confirms the transaction.
+3. The vigilante submits this proof to the Babylon chain using the `InsertBTCSpvProof` function. The proof is validated and stored in state. This includes parsing the submission, checking for duplicates, and verifying the checkpoint with the [checkpointing module](https://github.com/babylonlabs-io/babylon/blob/main/x/checkpointing/README.md).
```

Comment:
```suggestion
3. The [vigilante reporter](https://github.com/babylonlabs-io/vigilante/blob/47956edbb72112162e4cecca5b9d1e0ad840dd47/reporter/utils.go#L191) submits this proof to the Babylon chain using the `InsertBTCSpvProof` function. The proof is validated and stored in state. This includes parsing the submission, checking for duplicates, and verifying the checkpoint with the [checkpointing module](https://github.com/babylonlabs-io/babylon/blob/main/x/checkpointing/README.md).
```

Reply:
Maybe instead of "using the `InsertBTCSpvProof` function" -> invoking the Babylon RPC.

But just a nit again